subsys/bluetooth/host/conn.c

```diff
@@ -3733,6 +3749,22 @@ int bt_conn_le_create_synced(const struct bt_le_ext_adv *adv,
 	struct bt_conn *conn;
 	int err;
 
+	CHECKIF(ret_conn == NULL) {
+		return -EINVAL;
+	}
+
+	CHECKIF(*ret_conn != NULL) {
+		/** This warning is to inform the user that the connection reference is not NULL and
+		 * it may lead to a potential connection object leakage.
```

Comment:
```suggestion
		/* This warning is to inform the user that the connection reference is not NULL and
		 * it may lead to a potential connection object leakage.
```